### PR TITLE
Added NCM source

### PR DESCRIPTION
**Pull request by lervag**
The first commit adds a necovim source for nvim-completion-manager (as promised [here](https://github.com/roxma/nvim-completion-manager/issues/70#issuecomment-296418349)).

The second commit fixes a minor bug in a regexp, I think. Please review; if I'm mistaken I will of course update correspondingly.

### CODE REVIEW

File: autoload/cm/sources/necovim.vim [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"=============================================================================
+" FILE: necovim.vim (NCM source)
+" AUTHOR:  Shougo Matsushita <Shougo.Matsu@gmail.com>
+" License: MIT license
+"=============================================================================
+
+function! cm#sources#necovim#refresh(opt, ctx)
+  let l:startcol = necovim#get_complete_position(a:ctx.typed)
```
Comment (Shougo) [inline autoload/cm/sources/necovim.vim]: Please use `startcol` instead of `l:startcol`.

File: autoload/cm/sources/necovim.vim [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  let l:base = strpart(a:ctx.typed, l:startcol)
+  let l:cnd = necovim#gather_candidates(a:ctx.typed, l:base)
```
Comment (Shougo) [inline autoload/cm/sources/necovim.vim]: Same.

File: autoload/necovim.vim [hunk 3]
```diff
@@ -60,7 +60,7 @@ function! necovim#gather_candidates(input, complete_str) abort
       let keyword.abbr = prefix .
             \ get(keyword, 'abbr', keyword.word)[2:]
     endfor
-  elseif cur_text =~# '\<let\S'
+  elseif cur_text =~# '\<let\s'
```
Comment (Shougo) [inline autoload/necovim.vim:63]: Nice.

File: autoload/cm/sources/necovim.vim [hunk 4]
```diff
@@ -0,0 +1,12 @@
+"=============================================================================
+" FILE: necovim.vim (NCM source)
+" AUTHOR:  Shougo Matsushita <Shougo.Matsu@gmail.com>
```
Comment (Shougo) [inline autoload/cm/sources/necovim.vim]: Please add your name.

**Comment by Shougo** [comment]
I have commented.

File: autoload/cm/sources/necovim.vim (hunk 1 above)
Comment (lervag) [reply autoload/cm/sources/necovim.vim]: Just out of curiosity: Is there a reason not to be explicit about scope except subjective preference?

**Comment by lervag** [comment]
Thanks, I've updated the code to respect your comments.

File: autoload/cm/sources/necovim.vim (hunk 1 above)
Comment (Shougo) [reply autoload/cm/sources/necovim.vim]: This is my style.  And the scope is verbose and useless.

**Comment by Shougo** [comment]
Merged.

**Comment by lervag** [comment]
Thanks!

**Pull request merged**